packages/jest-config/src/normalize.js

```diff
@@ -4,192 +4,189 @@
+import type {InitialConfig} from 'types/Config';
 
+const {NODE_MODULES, DEFAULT_JS_PATTERN} = require('./constants');
+const {ValidationError, validate} = require('jest-validate');
 const chalk = require('chalk');
-const constants = require('./constants');
+const DEFAULT_CONFIG = require('./defaults');
 const path = require('path');
+const Resolver = require('jest-resolve');
 const utils = require('jest-util');
+const VALID_CONFIG = require('./validConfig');
+const DEPRECATED_CONFIG = require('./deprecated');
+
+const {
+  _replaceRootDirInPath,
+  _replaceRootDirTags,
+  getTestEnvironment,
+  resolve,
+} = require('./utils');
 
 const JSON_EXTENSION = '.json';
 const PRESET_NAME = 'jest-preset' + JSON_EXTENSION;
 
-const BULLET = chalk.bold('\u25cf ');
-const DEPRECATION_MESSAGE = `
-
-  Jest changed the default configuration for tests.
+const DOCUMENTATION_NOTE = `
 
-  ${chalk.bold('Configuration Documentation:')} https://facebook.github.io/jest/docs/configuration.html
-  ${chalk.bold('Jest Issue Tracker:')} https://github.com/facebook/jest/issues
+  ${chalk.bold('Configuration Documentation:')}
+  https://facebook.github.io/jest/docs/configuration.html
 `;
-const throwConfigurationError = message => {
-  throw new Error(chalk.red(message + DEPRECATION_MESSAGE));
-};
 
-const logConfigurationWarning = message => {
-  console.warn(chalk.yellow(BULLET + message + DEPRECATION_MESSAGE));
+const throwRuntimeConfigError = message => {
+  throw new ValidationError(
+    null,
+    '\n' + chalk.white(message) + DOCUMENTATION_NOTE
+  );
 };
 
-const resolve = (rootDir, key, filePath) => {
-  const module = Resolver.findNodeModule(
-    _replaceRootDirTags(rootDir, filePath),
+const setupPreset = (config: InitialConfig, configPreset: string) => {
+  let preset;
+  const presetPath = _replaceRootDirInPath(config.rootDir, configPreset);
+  const presetModule = Resolver.findNodeModule(
+    presetPath.endsWith(JSON_EXTENSION)
+      ? presetPath
+      : path.join(presetPath, PRESET_NAME),
     {
-      basedir: rootDir,
+      basedir: config.rootDir,
     },
   );
 
-  if (!module) {
-    throw new Error(
-      `Jest: Module "${filePath}" in the "${key}" option was not found.`
+  try {
+    // $FlowFixMe
+    preset = require(presetModule);
+  } catch (error) {
+    throw new ValidationError(
+      null,
+      chalk.white(`\n\n  Preset ${chalk.bold(presetPath)} not found.`)
     );
   }
 
-  return module;
+  if (config.setupFiles) {
+    config.setupFiles = preset.setupFiles.concat(config.setupFiles);
+  }
+  if (config.modulePathIgnorePatterns) {
+    config.modulePathIgnorePatterns = preset.modulePathIgnorePatterns
+      .concat(config.modulePathIgnorePatterns);
+  }
+  if (config.moduleNameMapper) {
+    config.moduleNameMapper = Object.assign(
+      {},
+      preset.moduleNameMapper,
+      config.moduleNameMapper,
+    );
+  }
+  // Don't show deprecation warnings if the setting comes from the preset.
```

Comment:
I think we can get rid of this if statement now. This was only a transitional issue between Jest and react-native that shouldn't come up any more.